src/main/java/games/strategy/engine/data/Resource.java

```diff
@@ -14,4 +19,14 @@
   public Resource(final String name, final GameData data) {
     super(name, data);
   }
+
+  public Resource(final String name, final GameData data, final List<PlayerID> players) {
+    this(name, data);
+    this.players = players;
+  }
+
+  public boolean isDisplayedFor(final PlayerID player) {
+    return players != null ? players.contains(player) : true;
```

Comment:
I believe it can be `null` when deserializing an old save game where `Resource#players` does not exist.  The alternative is to initialize the field to an empty list in a `readObject()` method.

If we're going to leave it nullable, then I'd recommend we add a TODO that says the `null` check can be removed in the next incompatible release, similar to what we've done in other domain classes when we've introduced this scenario.

Reply:
Correct. Null check is needed for old save games. I'll add a TODO to remove.

Reply:
Updated.